### PR TITLE
James 1901 Bis - Report metrics in ElasticSearch

### PR DESCRIPTION
**Pull request by Laura-Royet**
This PR is coming from: https://github.com/linagora/james-project/pull/541

### CODE REVIEW

File: server/container/metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/ESMetricReporter.java [hunk 1]
```diff
@@ -37,20 +37,24 @@
...
         if (esReporterConfiguration.isEnabled()) {
-            try {
-                this.reporter = Optional.of(ElasticsearchReporter.forRegistry(registry)
-                    .hosts(esReporterConfiguration.getHostWithPort())
-                    .index(esReporterConfiguration.getIndex())
-                    .build());
-            } catch (IOException e) {
-                throw Throwables.propagate(e);
-            }
+            this.reporter = getReporter(esReporterConfiguration, registry);
```
Comment (quynhn) [inline server/container/metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/ESMetricReporter.java]: It's better to move the if closure to extracted function I think.

File: server/container/metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/ESReporterConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ ****************************************************************/
+
+package org.apache.james.metrics.dropwizard;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+
+public class ESReporterConfiguration {
+
+    public static final boolean ENABLED = true;
```
Comment (strokyl) [inline server/container/metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/ESReporterConfiguration.java]: I don't really understand the necessity of those constant `ENABLED`, `DISABLED`, it seems overkill to me. I think passing directly `true` or `false` directly to the `ESReporterConfiguration` is clear enough.

File: server/container/metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/ESReporterConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    public ESReporterConfiguration(Optional<String> host, Optional<Integer> port, boolean enabled, Optional<String> index, Optional<Long> periodInSecond) {
+        this.host = host;
+        this.port = port;
+        this.enabled = enabled;
+        this.index = index;
+        this.periodInSecond = periodInSecond;
+    }
+
+    public String getHostWithPort() {
+        Preconditions.checkState(host.isPresent());
```
Comment (strokyl) [inline server/container/metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/ESReporterConfiguration.java]: `Preconditions.checkState(host.isPresent() && port.isPresent());` (just a suggestion, maybe I want to one line too much sometime).

File: server/container/metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/ESReporterConfiguration.java [hunk 4]
```diff
@@ -62,8 +62,7 @@ public ESReporterConfiguration(Optional<String> host, Optional<Integer> port, bo
     }
 
     public String getHostWithPort() {
-        Preconditions.checkState(host.isPresent());
-        Preconditions.checkState(port.isPresent());
+        Preconditions.checkState(host.isPresent() && port.isPresent());
```
Comment (rouazana) [inline server/container/metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/ESReporterConfiguration.java]: Sorry to answer on this a little late, but that's not our coding style. We prefer to split preconditions, so we can know exactly which one is failing.

Comment (strokyl) [inline server/container/metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/ESReporterConfiguration.java]: Sorry @Laura-Royet. @rouazana that's smart.

Comment (Laura-Royet) [inline server/container/metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/ESReporterConfiguration.java]: Ok, no problem.
Done :)

**Comment by rouazana** [comment]
It seems we are missing 2 things to merge this PR: integration tests and moving this to a new maven project.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**